src/applications/coronavirus-screener/lib/index.js

```diff
@@ -0,0 +1,59 @@
+import recordEvent from 'platform/monitoring/record-event';
+import moment from 'moment';
+import { scroller } from 'react-scroll';
+
+// return only enabled questions
+export function getEnabledQuestions(questionState) {
+  return questionState.filter(question => question.enabled ?? true);
+}
+
+// check result of answers
+export function checkFormResult(questionState) {
+  return getEnabledQuestions(questionState)
+    .map(question => {
+      const passValues = question.passValues ?? ['no'];
+      return passValues.includes(question.value);
+    })
+    .includes(false)
+    ? 'fail'
+    : 'pass';
+}
+
+// check if all enabled questions have been answered
+export function checkFormComplete(questionState) {
+  const completedQuestions = getEnabledQuestions(questionState).map(question =>
+    Object.prototype.hasOwnProperty.call(question, 'value'),
+  );
+  return !completedQuestions.includes(false);
+}
+
+// check the overall status of the form
+export function checkFormStatus(questionState) {
+  return !checkFormComplete(questionState)
+    ? 'incomplete'
+    : checkFormResult(questionState);
+}
+
+// record completion event to GA
+export function recordCompletion({ formState, setFormState }) {
+  if (formState.status !== 'incomplete' && formState.completed === false) {
```

Comment:
It seems like when this method is called formState is not yet updated so the analytic event is never fired.

Reply:
good catch, i'll take a look

Reply:
should be fixed now! lmk if you still see issues.